http/Router.cpp

```diff
@@ -48,17 +134,19 @@ bool DispatcherRoute::dispatch(const std::string& method, const std::string& mpa
     bool next = true;
 
     std::string cpath = path_concat(mpath, path);
-    /*
-    std::cout << "Dispatcher OriginalUrl: " << request.originalUrl << std::endl;
-    std::cout << "Dispatcher Path: " << path << std::endl;
-    std::cout << "Dispatcher MPath: " << mpath << std::endl;
-    std::cout << "Dispatcher CPath: " << cpath << std::endl;
-    */
-    if ((request.path.rfind(cpath, 0) == 0 && this->method == "use") || (cpath == request.path && (method == this->method || this->method == "all"))) {
-        request.url = request.originalUrl.substr(cpath.length());
+    
+    if ((request.path.rfind(cpath, 0) == 0 && this->method == "use") || (checkForUrlMatch(cpath, request.originalUrl) && (method == this->method || this->method == "all"))) {
+        // request.url = request.originalUrl.substr(cpath.length());
+        /* TODO: change to substr */
```

Comment:
we weren't totally sure why the substring is extracted here and it caused some problems because the length of cpath does not match the length of the parameterized URL. Since it was not used directly afterward we thought it does also work without creating the substring.

Reply:
If that's a question for me: Yes, req.url is currently not used anywhere in snode.c, so everything should work entirely without req.url and thus the assignment (substr substr or not) to req.url is currently unimportant . Then why does req.url even exist? Answer: req.url is a property of the request object in nodejs / express and is obviously also used internally there (https://expressjs.com/en/4x/api.html#req.originalUrl). As already noted, req.url is not yet used in snode.c. It is currently questionable whether it will ever be used.